### PR TITLE
feat(accessRight): render data depend on user type.

### PR DESCRIPTION
**Pull request by lalitaCF**

### CODE REVIEW

File: assets/app.js [hunk 1]
```diff
@@ -154,6 +154,7 @@ new Vue({
         Multiselect: window.VueMultiselect.default
     },
     data: {
+        accessRight:AccessRight,
```
Comment (cyberhck) [inline assets/app.js]: nit: space after Colon?

File: assets/app.js [hunk 2]
```diff
@@ -334,7 +335,8 @@ new Vue({
                 ]
             }
         },
-        translationUpdates: LogData.content
+        translationUpdates: LogData.content,
```
Comment (cyberhck) [inline assets/app.js]: nit: Was there a change here? You added a comma and an empty line.

File: assets/app.js [hunk 3]
```diff
@@ -684,6 +686,11 @@ new Vue({
             }
         }
     },
+    mounted:function(){
```
Comment (cyberhck) [inline assets/app.js]: nit: comma after Colon

File: assets/app.js [hunk 4]
```diff
@@ -922,6 +929,23 @@ new Vue({
                 api.call()
             }
         },
+        getDescriptorView: function (){
+
+            languages = this.accessRight;
+            AppLanguages.forEach(function (element) {
+                if (languages.includes(element['id'])) {
+                    element['status'] = true
```
Comment (cyberhck) [inline assets/app.js]: instead of this whole if statement, do a `element['status'] === languages.includes(element['id'])`?

Comment (lalitaCF) [reply assets/app.js]: @cyberhck 
you mean only this?
```
AppLanguages.forEach(function (element) {
     element['status'] === languages.includes(element['id')
});
``` 
This is not working.

Comment (paibamboo) [reply assets/app.js]: `element['status'] = languages.includes(element['id])`

Comment (cyberhck) [reply assets/app.js]: my bad, I forgot a `]`, can you try again?

Comment (cyberhck) [reply assets/app.js]: where's the `accessRight` defined? does `languages.includes` return `boolean`?

Comment (lalitaCF) [reply assets/app.js]: @paibamboo @cyberhck 
I try this it is work!
 ```
AppLanguages.forEach(function (element) {
       element['status'] = languages.includes(element['id'])
 });
```

- `where's the accessRight defined? `  it's defined in product_descriptor.html.twig in `shop`
- ` does languages.includes return boolean? ` yes

Comment (paibamboo) [reply assets/app.js]: By the way `.includes()` is not supported by IE. Will this app.js be run on browser or node environment? If on browser, I recommend to use `.indexOf()` instead.

Comment (cyberhck) [reply assets/app.js]: AFAIK we'll not be using IE, only chrome/firefox (backend), But I'm not sure 🙂

Comment (Solomonic) [reply assets/app.js]: We don't care for IE, since we know everyone using it. Nevertheless Pai is right: use `.indexOf()`

File: assets/app.js [hunk 5]
```diff
@@ -154,6 +154,7 @@ new Vue({
         Multiselect: window.VueMultiselect.default
     },
     data: {
+        accessRight: AccessRight,
```
Comment (Solomonic) [inline assets/app.js]: use 2 variables instead: 
- `isAdmin: true/false` and use `v-if="isAdmin"` as condition in the template
- get `AppLanguages` via php from the decoded json from `descriptor: {'is_admin':true/false, 'languages':[ {'id':'de', 'status': true, ...},{....}, ... ] }` 
where `language.id` is the unique language code, `language.status` describes if the language is active by default. `language.flag` should NOT be encoded in the access rights, this is a static css class which renders the language flag icon, but rather set when we set `AppLanguages`

File: assets/app.js [hunk 6]
```diff
@@ -684,6 +685,11 @@ new Vue({
             }
         }
     },
+    mounted: function(){
+        if(AccessRight != 1){
+            this.getDescriptorView()
+        }
+    },
```
Comment (Solomonic) [inline assets/app.js]: No need for doing that

File: assets/app.js [hunk 7]
```diff
@@ -922,6 +928,19 @@ new Vue({
                 api.call()
             }
         },
+        getDescriptorView: function (){
+
+            languages = this.accessRight;
```
Comment (Solomonic) [inline assets/app.js]: no type mix ups please .... same as discussed before

File: assets/app.js [hunk 8]
```diff
@@ -922,6 +928,19 @@ new Vue({
...
+            AppLanguages.forEach(function (element) {
+                element['status'] === languages.includes(element['id'])
+            });
+
+            this.show_load = false
+            this.isFullScreen = true
+            this.show_translations = true
+
+            window.scrollTo(0, 0)
+        },
```
Comment (Solomonic) [inline assets/app.js]: I don't think we need this method. To set the proper display options and default headline for Translators rather do in `data`:
- `show_names: this.isAdmin`
- `headline: this.defaultHeadline`
- `show_load: !this.isAdmin`
- `isFullScreen: !this.isAdmin`
and in `computed`
- `defaultHeadline: function(){ if (this.isAdmin) return "Product Names" else return "Translation Management"}`

File: components/templates/template.html.twig [hunk 9]
```diff
@@ -1,818 +1,2350 @@
...
+                                           @input="stashUpdate('metatags', cell)"
+                                           @keyup.enter="saveTranslationUpdate('metatags', cell)"/>
+                                </div>
+                            </template>
+                            <template slot="cs" scope="cell">
+                                <div v-if="cell.row['is_active']=='1'">
+                                    <span class="absolute flag-icon flag-icon-cz"></span>
+                                    <input class="table_input" v-model:value="cell.row['cs']" type="text"
+                                           @input="stashUpdate('metatags', cell)"
+                                           @keyup
```
Comment (Solomonic) [inline components/templates/template.html.twig]: Why is basically all the template changed? you should have minimal changes in here, can you please check if you changed the spaces to tabs or the encoding. A proper review is like that hardly possible

**Review by Solomonic — CHANGES_REQUESTED** [review]
- introduce isAdmin: true/false
- generate AppLanguages from descriptor.languages
- ensure that only your real template changes are committed

**Pull request closed without merging**